Mathlib/Computability/ContextFreeGrammar.lean

```diff
@@ -253,3 +254,447 @@ theorem Language.IsContextFree.reverse (L : Language T) :
   fun ⟨g, hg⟩ => ⟨g.reverse, hg ▸ Set.ext g.mem_reverse_language_iff_reverse_mem_language⟩
 
 end closure_reversal
+
+section lift_sink
+
+/- This section contains only auxiliary constructions that will shorten upcoming proofs of
+closure properties. When combining several grammars together, we usually want to take a sum type of
+their nonterminal types and lift respective nonterminals to this sum type. We subsequently show
+that the resulting grammar preserves derivations of those strings that may contain any terminals but
+only the proper nonterminals. The lifting operation must be injective. The sinking operation must be
+injective on those symbols where it is defined. -/
+
+/-- Lifting `Symbol` to a larger nonterminal type. -/
+def Symbol.map {N₀ N : Type*} (liftN : N₀ → N) : Symbol T N₀ → Symbol T N
```

Comment:
Even if that's how you use it, there is no notion of `N` being bigger than `N₀` here, right? Also this is not a lifting operation. Mind clarifying the docstring?

Reply:
I mean, I require injective map from `N₀` to `N` so we can say that `N` is bigger. However, I welcome your suggestions for reformulation.

Reply:
Where do you require that it's injective? Certainly not in the definition